src/App.tsx

```diff
@@ -1,19 +1,139 @@
 import React from 'react';
 import './App.css';
-
+import todos from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
 import users from './api/users';
+import { State } from './components/types/State';
+
+const preparedTodos = todos.map((el) => ({
+  ...el,
+  user: users.find(
+    (user) => user.id === el.userId,
+  ) || null,
+}));
+
+class App extends React.Component<{}, State> {
+  state:State = {
+    title: '',
+    userName: '',
+    todoList: [...preparedTodos],
+    formErrors: { title: '', user: '' },
+  };
+
+  onTitleChange = (event:React.ChangeEvent<HTMLInputElement>) => {
+    this.setState(state => ({
+      title: event.target.value,
+      formErrors: { ...state.formErrors, title: '' },
+    }));
+  };
+
+  onSelectorChange = (event:React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState(state => ({
+      userName: event.target.value,
+      formErrors: { ...state.formErrors, user: '' },
+    }));
+  };
+
+  onAdd = () => {
+    const newTodo = {
+      userId: Number(this.state.userName),
```

Comment:
if 2 users have the same name - their id will not be unique. You should find User object in array and write here their unique id